### ISSUE TITLE
API Platform supported ?

### ISSUE THREAD

**Issue by Romaixn** [knpuniversity/oauth2-client-bundle#376]
Hello everyone !

Is that possible to integrate social oauth in API Platform ? Like a "/oauth/login/google" route which render a JWT Token for user for example ?

I have never done this kind of thing!

Thanks for your help,

Cheers

**Comment by weaverryan** [comment]
I'm not entirely sure... it depends on what flow you have in mind. This bundle handles the "authorization code" grant type for OAuth. This is where you click a "login with google" link on your site, then the browser is fully redirected to a URL over on Google (to authorize your app) and then you are fully redirected back to your site. If you're building an SPA, I think this flow may be slightly different - but this is not something I have experience with - but you can see https://www.oauth.com/oauth2-servers/single-page-apps/

So, in short - I'm not entirely sure :). If you're building an API for your own frontend to consume, then sure - you could use the authorization code flow to authenticate the user because, in addition to being an API, your backend could also support receiving the redirect from google, exchanging the code for the access token, and authenticating the user (which could be stored in the session like normal or return a JWT if you want).

Sorry I can't be more clear - it depends on your setup and my knowledge gets limited towards SPAs.

**Comment by jraller** [comment]
@Romaixn I am building an API Platform solution where I use Symfony firewall rules to allow both username and password login and also CAS authentication using hwi/HWIOAuthBundle (but there are issues with the latest release) and for both of these I return a JWT to the user and authenticate all further interactions using that JWT. I'm browsing the issues here because I'm wondering if I should switch to the knpunivesity/oauth2-client-bundle instead. The tricks for me were altering the Caddyfile to allow the login and connect routes to pass, and the configuration of my OAuth2, and the addition of a controller to accept the OAuth return and issue a JWT.
